### ISSUE TITLE
helm 3 alpha 1: helm chart fails badly when chart path is invalid

### ISSUE THREAD

**Issue by willholley** [helm/helm#5776]
Output of `helm version`:
`﻿﻿version.BuildInfo{Version:"v3.0.0-alpha.1", GitCommit:"b9a54967f838723fe241172a6b94d18caf8bcdca", GitTreeState:"clean"}`

Attempting to push a chart with a badly formed ref fails with a stack trace. Ideally this error would be a bit more friendly and/or could be prevented at `helm chart save`.

```
helm chart save mychart mychart

Using default tag: latest
32e5dd9: Saving meta (146 B)
5282241: Saving content (3.1 KiB)
Name: mychart
Version: 1.0.8
Meta: sha256:32e5dd94fcba61d0bc36664fc0f9e908e39552507f7694c21f7a34f19eb82572
Content: sha256:5282241bf2a576f84ab38c681cf325ad6f03d35100f76bfb31be119d86f9d675
latest: saved

helm chart push mychart
Using default tag: latest
The push refers to repository [mychart]
Name: mychart
Version: 1.0.8
Meta: sha256:32e5dd94fcba61d0bc36664fc0f9e908e39552507f7694c21f7a34f19eb82572
Content: sha256:5282241bf2a576f84ab38c681cf325ad6f03d35100f76bfb31be119d86f9d675
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
helm.sh/helm/vendor/github.com/containerd/containerd/reference.Spec.Hostname(0xc0006e1490, 0x6, 0x0, 0x0, 0xc0006e1490, 0x7)
	/go/src/helm.sh/helm/vendor/github.com/containerd/containerd/reference/reference.go:129 +0x87
helm.sh/helm/vendor/github.com/containerd/containerd/remotes/docker.(*dockerResolver).base(0xc000629900, 0xc0006e1490, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/helm.sh/helm/vendor/github.com/containerd/containerd/remotes/docker/resolver.go:307 +0x88
helm.sh/helm/vendor/github.com/containerd/containerd/remotes/docker.(*dockerResolver).Pusher(0xc000629900, 0x270f6e0, 0xc0006e7b30, 0xc0006e1329, 0x6, 0x0, 0x0, 0xc0006bc810, 0x0)
	/go/src/helm.sh/helm/vendor/github.com/containerd/containerd/remotes/docker/resolver.go:281 +0x158
helm.sh/helm/vendor/github.com/deislabs/oras/pkg/oras.Push(0x270f6e0, 0xc0006e7b30, 0x270b1e0, 0xc0006e2ae0, 0xc0006e1329, 0x6, 0x26d9760, 0xc0006e40c0, 0xc0006f55e0, 0x2, ...)
	/go/src/helm.sh/helm/vendor/github.com/deislabs/oras/pkg/oras/push.go:36 +0x246
helm.sh/helm/pkg/registry.(*Client).PushChart(0xc000629940, 0xc0006e7aa0, 0xc0006e7aa0, 0x0)
	/go/src/helm.sh/helm/pkg/registry/client.go:100 +0x2fd
helm.sh/helm/pkg/action.(*ChartPush).Run(0xc000761d70, 0x26dbee0, 0xc0000ce000, 0x7ffeefbfe7d9, 0x6, 0xc000761d30, 0x0)
	/go/src/helm.sh/helm/pkg/action/chart_push.go:43 +0x7b
main.newChartPushCmd.func1(0xc000434c80, 0xc0006e2a70, 0x1, 0x1, 0x0, 0x0)
	/go/src/helm.sh/helm/cmd/helm/chart_push.go:44 +0x74
helm.sh/helm/vendor/github.com/spf13/cobra.(*Command).execute(0xc000434c80, 0xc0006e2a30, 0x1, 0x1, 0xc000434c80, 0xc0006e2a30)
	/go/src/helm.sh/helm/vendor/github.com/spf13/cobra/command.go:762 +0x465
helm.sh/helm/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc00057c280, 0x26dbee0, 0xc0000ce000, 0xc0000d2010)
	/go/src/helm.sh/helm/vendor/github.com/spf13/cobra/command.go:852 +0x2ec
helm.sh/helm/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/helm.sh/helm/vendor/github.com/spf13/cobra/command.go:800
main.main()
	/go/src/helm.sh/helm/cmd/helm/helm.go:55 +0xa6
```

**Comment by willholley** [comment]
@jdolitsky yes, that sounds right. Looking briefly at the [OCI spec](https://github.com/opencontainers/distribution-spec/blob/master/spec.md#pulling-an-image), both `helm chart save` and `helm chart push` should validate that `<ref>` is a valid url, which may optionally include a digest or tag.

**Comment by jdolitsky** [comment]
@willholley I'm not sure if I agree this should be required for `helm chart save`. I see this as sort of the equivalent of `docker tag`. There are several issues surrounding this UX we need to address, please join us on the dev calls Thursday 9:30am PST

**Comment by bacongobbler** [comment]
@jdolitsky do you think this use case has been fixed, or is this still an issue in core?

**Comment by jdolitsky** [comment]
@bacongobbler just tested it out, this issue is still present on push

**Comment by jdolitsky** [comment]
This issue appears to be fixed in Helm 3.0.0+:

```
$ helm chart push mychart
Error: Chart not found: mychart
```
